netty-tcnative-boringssl-static-2.0.76.Final-linux-aarch_64 crash

Hello,

The latest version, `2.0.76.Final` of `netty-tcnative-boringssl-static` always crashes for us on Linux ARM (`linux-aarch_64` classifier).
We're using netty `4.2.12.Final`.

On AL2023 with java-25-amazon-corretto-headless:

```
java.lang.Exception: No such file or directory
        at io.netty.internal.tcnative.SSLContext.make(Native Method)
        at io.netty.handler.ssl.ReferenceCountedOpenSslContext.<init>(ReferenceCountedOpenSslContext.java:325)
        ... 42 common frames omitted
Wrapped by: javax.net.ssl.SSLException: failed to create an SSL_CTX
        at io.netty.handler.ssl.ReferenceCountedOpenSslContext.<init>(ReferenceCountedOpenSslContext.java:327)
```

On DockerDesktop on MacOS ARM (hence a Linux VM):

```
java.lang.Exception: Invalid argument
        at io.netty.internal.tcnative.SSLContext.make(Native Method)
        at io.netty.handler.ssl.ReferenceCountedOpenSslContext.<init>(ReferenceCountedOpenSslContext.java:325)
        ... 42 common frames omitted
Wrapped by: javax.net.ssl.SSLException: failed to create an SSL_CTX
        at io.netty.handler.ssl.ReferenceCountedOpenSslContext.<init>(ReferenceCountedOpenSslContext.java:327)
```

Other flavors, including `linux-x86_64` and `osx-aarch_64` work as expected.
`2.0.75.Final` works as expected.

Please let me know how we can help investigating this.

Dockerfile:

```dockerfile
FROM amazonlinux:2023

RUN dnf install -y java-25-amazon-corretto-devel maven git

RUN git clone https://github.com/gatling/netty-boringssl-bug2.git

ENV JAVA_HOME=/etc/alternatives/java_sdk_25

WORKDIR /netty-boringssl-bug2

CMD ["/usr/bin/mvn", "compile", "exec:exec"]
```


test with:
```console
$ docker build --tag=netty-boringssl-bug2 .
$ docker run --rm -it netty-boringssl-bug2:latest
```

When on x86: `BUILD SUCCESS`
When on ARM: 
```
[WARN ] i.n.h.s.OpenSsl - Failed to get the list of available OpenSSL cipher suites.
java.lang.Exception: No such file or directory
        at io.netty.internal.tcnative.SSLContext.make(Native Method)
        at io.netty.handler.ssl.OpenSsl.<clinit>(OpenSsl.java:210)
        at io.netty.handler.ssl.SslContext.newClientContextInternal(SslContext.java:898)
        at io.netty.handler.ssl.SslContextBuilder.build(SslContextBuilder.java:764)
        at io.gatling.netty.ClientSslContextBug.main(ClientSslContextBug.java:8)
Exception in thread "main" javax.net.ssl.SSLException: failed to create an SSL_CTX
        at io.netty.handler.ssl.ReferenceCountedOpenSslContext.<init>(ReferenceCountedOpenSslContext.java:327)
        at io.netty.handler.ssl.OpenSslContext.<init>(OpenSslContext.java:40)
        at io.netty.handler.ssl.OpenSslClientContext.<init>(OpenSslClientContext.java:197)
        at io.netty.handler.ssl.SslContext.newClientContextInternal(SslContext.java:899)
        at io.netty.handler.ssl.SslContextBuilder.build(SslContextBuilder.java:764)
        at io.gatling.netty.ClientSslContextBug.main(ClientSslContextBug.java:8)
Caused by: java.lang.Exception: No such file or directory
        at io.netty.internal.tcnative.SSLContext.make(Native Method)
        at io.netty.handler.ssl.ReferenceCountedOpenSslContext.<init>(ReferenceCountedOpenSslContext.java:325)
        ... 5 more
[ERROR] Command execution failed.
org.apache.commons.exec.ExecuteException: Process exited with an error: 1 (Exit value: 1)
    at org.apache.commons.exec.DefaultExecutor.executeInternal (DefaultExecutor.java:394)
    at org.apache.commons.exec.DefaultExecutor.execute (DefaultExecutor.java:292)
    at org.codehaus.mojo.exec.ExecMojo.executeCommandLine (ExecMojo.java:913)
    at org.codehaus.mojo.exec.ExecMojo.executeCommandLine (ExecMojo.java:873)
    at org.codehaus.mojo.exec.ExecMojo.execute (ExecMojo.java:464)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:972)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:293)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:196)
    at jdk.internal.reflect.DirectMethodHandleAccessor.invoke (DirectMethodHandleAccessor.java:104)
    at java.lang.reflect.Method.invoke (Method.java:565)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
```

--- a/boringssl-static/pom.xml
+++ b/boringssl-static/pom.xml
@@ -717,6 +717,22 @@
       </properties>
 
       <build>
+        <pluginManagement>
+          <plugins>
+            <plugin>
+              <artifactId>maven-enforcer-plugin</artifactId>
+              <version>1.4.1</version>
+              <dependencies>
+                <!-- Provides the 'requireFilesContent' enforcer rule. -->
+                <dependency>
+                  <groupId>com.ceilfors.maven.plugin</groupId>
+                  <artifactId>enforcer-rules</artifactId>
+                  <version>1.2.0</version>
+                </dependency>
+              </dependencies>
+            </plugin>
+          </plugins>
+        </pluginManagement>
         <plugins>
           <plugin>
             <artifactId>maven-enforcer-plugin</artifactId>
@@ -735,6 +751,15 @@
                       <property>os.detected.classifier</property>
                       <regex>^linux-x86_64.*</regex>
                     </requireProperty>
+                    <requireFilesContent>
+                      <message>
+                        Cross compile and Release process must be performed on RHEL 7.6 or its derivatives.
+                      </message>
+                      <files>
+                        <file>/etc/redhat-release</file>
+                      </files>
+                      <content>release 7.6</content>
+                    </requireFilesContent>
                   </rules>
                   <ignoreCache>true</ignoreCache>
                 </configuration>

--- a/docker/Dockerfile.cross_compile_aarch64
+++ b/docker/Dockerfile.cross_compile_aarch64
@@ -1,4 +1,4 @@
-FROM --platform=linux/amd64 centos:7.9.2009
+FROM --platform=linux/amd64 centos:7.6.1810
 
 ARG gcc_version=10.2-2020.11
 ARG openssl_version=1_1_1d
@@ -15,7 +15,10 @@ RUN mkdir $SOURCE_DIR
 WORKDIR $SOURCE_DIR
 
 # Update to use the vault
-RUN sed -i -e 's/^mirrorlist/#mirrorlist/g' -e 's/^#baseurl=http:\/\/mirror.centos.org\/centos\/$releasever\//baseurl=https:\/\/linuxsoft.cern.ch\/centos-vault\/\/7.9.2009\//g' /etc/yum.repos.d/CentOS-Base.repo
+RUN sed -i -e 's/^mirrorlist/#mirrorlist/g' -e 's/^#baseurl=http:\/\/mirror.centos.org\/centos\/$releasever\//baseurl=https:\/\/linuxsoft.cern.ch\/centos-vault\/\/7.6.1810\//g' /etc/yum.repos.d/CentOS-Base.repo
+
+# We want to have git 2.x for the maven scm plugin and also for boringssl
+RUN yum install -y https://opensource.blueoptima.com/centos/6/git/x86_64/wandisco-git-release-6-1.noarch.rpm
 
 # Install requirements
 RUN  set -x && \

--- a/openssl-dynamic/pom.xml
+++ b/openssl-dynamic/pom.xml
@@ -345,6 +345,18 @@
                 </archive>
               </configuration>
             </plugin>
+            <plugin>
+              <artifactId>maven-enforcer-plugin</artifactId>
+              <version>1.4.1</version>
+              <dependencies>
+                <!-- Provides the 'requireFilesContent' enforcer rule. -->
+                <dependency>
+                  <groupId>com.ceilfors.maven.plugin</groupId>
+                  <artifactId>enforcer-rules</artifactId>
+                  <version>1.2.0</version>
+                </dependency>
+              </dependencies>
+            </plugin>
           </plugins>
         </pluginManagement>
 
@@ -366,6 +378,15 @@
                       <property>os.detected.classifier</property>
                       <regex>^linux-x86_64.*</regex>
                     </requireProperty>
+                    <requireFilesContent>
+                      <message>
+                        Cross compile and Release process must be performed on RHEL 7.6 or its derivatives.
+                      </message>
+                      <files>
+                        <file>/etc/redhat-release</file>
+                      </files>
+                      <content>release 7.6</content>
+                    </requireFilesContent>
                     <requireProperty>
                       <property>aprArmHome</property>
                       <message>The folder of APR for aarch64 must be specified by hand. Please try -DaprArmHome=</message>

--- a/pom.xml
+++ b/pom.xml
@@ -75,7 +75,7 @@
   <properties>
     <maven.javadoc.failOnError>false</maven.javadoc.failOnError>
     <checkstyle.skip>true</checkstyle.skip>
-    <enforcer.plugin.version>3.0.0</enforcer.plugin.version>
+    <enforcer.plugin.version>1.4.1</enforcer.plugin.version>
     <maven.compiler.source>1.8</maven.compiler.source>
     <maven.compiler.target>1.8</maven.compiler.target>
     <maven.compiler.release>8</maven.compiler.release>
@@ -161,6 +161,14 @@
         <plugin>
           <artifactId>maven-enforcer-plugin</artifactId>
           <version>${enforcer.plugin.version}</version>
+          <dependencies>
+            <!-- Provides the 'requireFilesContent' enforcer rule. -->
+            <dependency>
+              <groupId>com.ceilfors.maven.plugin</groupId>
+              <artifactId>enforcer-rules</artifactId>
+              <version>1.2.0</version>
+            </dependency>
+          </dependencies>
         </plugin>
         <plugin>
           <groupId>org.fusesource.hawtjni</groupId>
@@ -839,10 +847,27 @@
                       <property>os.detected.classifier</property>
                       <regex>^linux-x86_64$</regex>
                     </requireProperty>
+                    <requireFilesContent>
+                      <message>
+                        Release process must be performed on RHEL 6.8 or its derivatives.
+                      </message>
+                      <files>
+                        <file>/etc/redhat-release</file>
+                      </files>
+                      <content>release 6.</content>
+                    </requireFilesContent>
                   </rules>
                 </configuration>
               </execution>
             </executions>
+            <dependencies>
+              <!-- Provides the 'requireFilesContent' enforcer rule. -->
+              <dependency>
+                <groupId>com.ceilfors.maven.plugin</groupId>
+                <artifactId>enforcer-rules</artifactId>
+                <version>1.2.0</version>
+              </dependency>
+            </dependencies>
           </plugin>
         </plugins>
       </build>

